picky/src/jose/jwe.rs

```diff
@@ -168,6 +201,112 @@ pub enum JweAlg {
     EcdhEsAesKeyWrap256,
 }
 
+#[derive(Debug, Clone, Copy)]
+enum KeyWrappingAlg {
+    Aes128,
+    Aes192,
+    Aes256,
+}
+
+impl KeyWrappingAlg {
+    fn key_size(self) -> usize {
+        match self {
+            KeyWrappingAlg::Aes128 => 16,
+            KeyWrappingAlg::Aes192 => 24,
+            KeyWrappingAlg::Aes256 => 32,
+        }
+    }
+
+    /// Decrypts wrapped CEK using the given AES decryption key
+    ///
+    /// SAFETY:
+    ///   - Caller must unsure `decryption_key` size matches the wrapping algorithm
+    fn decrypt_key(
+        &self,
+        cek_alg: JweEnc,
+        encrypted_cek: &[u8],
+        decryption_key: &[u8],
+    ) -> Result<Zeroizing<Vec<u8>>, JweError> {
```

Comment:
I guess this function is not really "unsafe" like in "memory safety".
Do you think you could clarify what happens when `decryption_key` size does not matches the wrapping algorithm? Is it returning an `Err`? Does it cause a panic? A misbehavior?

In any cases, "Safety" sections are typically used to document invariants that must be uphold when calling an unsafe function.

For reference, the [API guidelines book](https://rust-lang.github.io/api-guidelines/documentation.html#function-docs-include-error-panic-and-safety-considerations-c-failure) says:

> Unsafe functions should be documented with a "Safety" section that explains all invariants that the caller is responsible for upholding to use the function correctly.

Maybe calling this section "Invariant" can help clarify to the reader that there is no hidden "memory safety"  considerations to have with this function:
```suggestion
    /// Decrypts wrapped CEK using the given AES decryption key
    ///
    /// # Invariants
    ///
    /// - Caller must unsure `decryption_key` size matches the wrapping algorithm
    fn decrypt_key(
        &self,
        cek_alg: JweEnc,
        encrypted_cek: &[u8],
        decryption_key: &[u8],
    ) -> Result<Zeroizing<Vec<u8>>, JweError> {
```

Reply:
Well, it will panic (`GenericArray` will throw it). Agreed, `Invariant` will sound better!

Reply:
done

Reply:
Oh, thank you!